### ISSUE TITLE
Remove effectivly empty CONTRIBUTING files

### ISSUE THREAD

**Issue by HannesWell** [eclipse-platform/eclipse.platform.releng.aggregator#2448]
Let's make sure issue is not already fixed in latest builds first.

- [x] I verified [latest Integration Build of Eclipse SDK](https://download.eclipse.org/eclipse/downloads/) doesn't provide the feature.

## Suggestion

Currently many repositories for Eclipse-Platform, Equinox, JDT and PDE have have `CONTRIBUTING` or a `CONTRIBUTING.md` file or both that most of the time just contains a link to the contribution information maintained in the corresponding organization's `.github` repository.

For example in the eclipse.platform repository
- https://github.com/eclipse-platform/eclipse.platform/blob/7fa51b7a44ee2074b93eaeed1c8bea13be59e541/CONTRIBUTING
- https://github.com/eclipse-platform/eclipse.platform/blob/7fa51b7a44ee2074b93eaeed1c8bea13be59e541/CONTRIBUTING.md

That link is then also often repeated in a Contributing section in the README.md of the repository.
See for example
- https://github.com/eclipse-platform/eclipse.platform?tab=readme-ov-file#how-to-contribute

With that the content of the `CONTRIBUTING` seems to be duplicated and actually not necessary.
Therefore I propose to delete the CONTRIBUTING if the scenario is as described above.

If I read the `Eclipse Foundation Project Handbook` correct, having a `CONTRIBUTING` is not mandatory per project and contributing information can also be contained in the README:
- https://www.eclipse.org/projects/handbook/#legaldoc-contributor

@waynebeaton is that correct or are we not allowed to perform the proposed task?

The contribution information generated by GitHub for a repository inherits the  `CONTRIBUTING` file/information from the organization if no such file exists in the repository. For the `jdt.debug` repository this proposal was already applied and it worked as expected there:
- https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/532#issuecomment-2405723053

See https://github.com/eclipse-jdt/eclipse.jdt.debug/contribute

On the other hand https://github.com/eclipse-jdt/eclipse.jdt.debug/contribute currently needs one more click to reach the final destination.

## Community

- [x] I understand suggesting an enhancement doesn't mandate anyone to implement it. Other contributors may consider this suggestion, or not, at their own convenience. The most efficient way to get it fixed is that I implement it myself and contribute it back as a good quality patch to the project.

**Comment by HannesWell** [comment]
@waynebeaton would this change be fine from a legal l respectively project rules point of view?

**Comment by HannesWell** [comment]
If there is no objection, especially by @waynebeaton (I think developers don't insist on an almost empty file), I'll start implementing this soon.

**Comment by waynebeaton** [comment]
IMHO a reasonable argument can be made that the CONTRIBUTING file is a contextual artifact; that is, it describes the process for working with a repository while that repository is in the context of an Eclipse project's GitHub org. Since the .github repository includes a relevant CONTRIBUTING file, we can assert that the requirement is met.

Having said all that, I invite you to carefully consider what this means for potential forks and whether or not the entire team is comfortable with this information being disconnected from the repository (that is, absent from forks).

**Comment by HannesWell** [comment]
Thank you.

> Having said all that, I invite you to carefully consider what this means for potential forks and whether or not the entire team is comfortable with this information being disconnected from the repository (that is, absent from forks).

Absolutely, at the moment I only have one opinion from https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/532#issuecomment-2406016063

But I encourage all @eclipse-platform/eclipse-platform-committers to express their opinion, especially if they are against this change. The same applies for all Equinox, PDE and JDT committers (unfortunately I cannot link them here easily, but (almost) all Equinox and PDE committers are also platform committers).

**Comment by Bananeweizen** [comment]
> is there a **default** link to "contrubite" file somewhere in github UI

The only one I'm aware of is "Insights" (at the top) > Community Standards > Contributing. And that one seems to work fine with the changes of Hannes.

@HannesWell You are aware of the Otterdog automation stuff, right? I'm not sure if Otterdog is already doing something to CONTRIBUTION or whether something is planned. But it would be good, if your strategy fits to whatever happens there, otherwise this might go back and forth.

**Comment by laeubi** [comment]
> I'm maybe blind, but is there a **default** link to "contrubite" file somewhere in github UI (not im our readme),

If you open an issue for example you have
![grafik](https://github.com/user-attachments/assets/2cda7502-efd3-42aa-82cd-0ed9bdef76cb)

there are similar links when one creates a PR.

**Comment by iloveeclipse** [comment]
Yeah, but that is not what I meant. I meant, on the **first** project page, there is nothing about "contributing"... 
Really strange.

**Comment by BeckerWdf** [comment]
So do it understand it correctly.
You want to remove all the contribution files that simply link to https://github.com/eclipse-platform/.github/blob/main/CONTRIBUTING.md?

But would don't plan to delete https://github.com/eclipse-platform/.github/blob/main/CONTRIBUTING.md itself?

My only concern is that if contributors arrive at e.g. the platform project they don't find a link to contribution information. So we should still have an entry point to that information in the single orgs/repos.

**Comment by HeikoKlare** [comment]
I am fine with the proposed changes. Since GitHub conventions allow to just have one central contributing document in the `.github` repository of an organization, the only remaining scenario I see where people may search for this information is when they first come to a project, as also mentioned by Matthias:

> My only concern is that if contributors arrive at e.g. the platform project they don't find a link to contribution information. So we should still have an entry point to that information in the single orgs/repos.

In this case, I would expect that people first look at information in the readme of a repository. And right now, at least in the Platform repository, we have this information in the readme: https://github.com/eclipse-platform/eclipse.platform?tab=readme-ov-file#how-to-contribute
We may check that this is the case for all repositories, but if we have that, I would consider this sufficient.

**Comment by BeckerWdf** [comment]
> In this case, I would expect that people first look at information in the readme of a repository. And right now, at least in the Platform repository, we have this information in the readme: https://github.com/eclipse-platform/eclipse.platform?tab=readme-ov-file#how-to-contribute
> We may check that this is the case for all repositories, but if we have that, I would consider this sufficient.

That sounds good to me.

**Comment by HannesWell** [comment]
> > My only concern is that if contributors arrive at e.g. the platform project they don't find a link to contribution information. So we should still have an entry point to that information in the single orgs/repos.
> 
> In this case, I would expect that people first look at information in the readme of a repository. And right now, at least in the Platform repository, we have this information in the readme: https://github.com/eclipse-platform/eclipse.platform?tab=readme-ov-file#how-to-contribute We may check that this is the case for all repositories, but if we have that, I would consider this sufficient.

Yes that was absolutely my intention, sorry if this wasn't clear in the initial text.
My plan would be to have a relativly prominent `CONTRIBUTING` section (exact name can vary) in the README of each repsitory, which links to the organization's `CONTRIBUTING.md` file and also contains the 'button' for the Oomph setup, similar to how it's in
https://github.com/eclipse-jdt/eclipse.jdt.debug/?tab=readme-ov-file#contributing or for eclipse.platform as Heiko linked it.

> I'm maybe blind, but is there a **default** link to "contrubite" file somewhere in github UI (not im our readme), similar to https://github.com/eclipse-jdt/eclipse.jdt.debug "code of conduct", "license", "security" etc links both on the right site and the bottom of the page?

You can append `/contribute` to a repository URL to get basic contributing information like good-first-issues and a link to the repositories or Organization's CONTRIBUTING file. For example:
- https://github.com/eclipse-jdt/eclipse.jdt.debug/contribute (references Orga's contributing file)
- https://github.com/eclipse-platform/eclipse.platform/contribute (references repos contributing file)

Unfortunately I haven't found it in the standard GH UI yet, I only know it from Jörg.
